### ISSUE TITLE
Cannot load alexnet.baseline.pth.tar 

### ISSUE THREAD

**Issue by JiangPQ** [jiecaoyu/XNOR-Net-PyTorch#14]
I use my own code to load the 'alexnet.baseline.pth.tar' file.
However, the keys in the file are not consistent with those in the model which are shown in the pic.

![screenshot from 2018-04-11 09-43-27](https://user-images.githubusercontent.com/10220802/38592327-026713c0-3d6e-11e8-8eda-df89f6029af5.png)

And the model's code is 

    class AlexNet(nn.Module):
        def __init__(self, num_classes=1000):
            super(AlexNet, self).__init__()
            self.num_classes = num_classes
            self.features = nn.Sequential(
                nn.Conv2d(3, 96, kernel_size=11, stride=4, padding=2),
                nn.BatchNorm2d(96, eps=1e-4, momentum=0.1, affine=True),
                nn.ReLU(inplace=True),
                nn.MaxPool2d(kernel_size=3, stride=2),
                BinConv2d(96, 256, kernel_size=5, stride=1, padding=2, groups=1),
                nn.MaxPool2d(kernel_size=3, stride=2),
                BinConv2d(256, 384, kernel_size=3, stride=1, padding=1),
                BinConv2d(384, 384, kernel_size=3, stride=1, padding=1, groups=1),
                BinConv2d(384, 256, kernel_size=3, stride=1, padding=1, groups=1),
                nn.MaxPool2d(kernel_size=3, stride=2),
            )
            self.classifier = nn.Sequential(
                BinConv2d(256 * 6 * 6, 4096, Linear=True),
                BinConv2d(4096, 4096, dropout=0.5, Linear=True),
                nn.BatchNorm1d(4096, eps=1e-3, momentum=0.1, affine=True),
                nn.Dropout(),
                nn.Linear(4096, num_classes),
            )

Shouldn't it be 'features.0.weight' but rather 'features.module.0.weight' ? 
Or am I using a wrong pretrained model?
BTW I'm really curious how did you attach the '.module'? LOL

**Comment by jiecaoyu** [comment]
@JiangPQ Hi, I think it is because I am using multiple GPUs. Using the ```torch.nn.DataParallel()``` will add the '.module'. I will have a further test after finishing my current experiment.

I think you are using the right model. My suggestion is to remove '.module' from the keys. It should be fine. I forget to remove the '.module' before saving the model.

**Comment by JiangPQ** [comment]
Thank you!
Here's my solution for people who met the same problem as me:
    
    def alexnet(pretrained=True, **kwargs):
 
        model = AlexNet(**kwargs)
        if pretrained:
            model_path = './alexnet.baseline.pth.tar'
            pretrained_model = torch.load(model_path)['state_dict']
            dict_content = [(k.replace('module.', ''), v) for k, v in pretrained_model.items()]
            pretrained_model = OrderedDict(dict_content)
            print([k for k, v in pretrained_model.items()])
            model.load_state_dict(pretrained_model)
        return model
